design-proposals/multus-annotation-customization.md

```diff
@@ -0,0 +1,59 @@
+# Overview
+Providing a way to customize the Multus pod annotation generated by the `virt-controller` where only a subset of the 
+Multus contract is currently mapped:
+- Name
+- Namespace
+- Interface name
+- MAC Address (if set in `interface` definition)
+
+## Motivation
+Improving the Multus support. As an example, customizing the IP through the annotation avoids having inconsistency between
+the IP dynamically assigned by Multus daemonset to the pod and the one defined guest level.
+
+## Goals
+- Consistency and visibility by having pod level what is defined in the guest
+- [Multus service](https://github.com/k8snetworkplumbingwg/multus-service) could work as expected by 
+having the Multus endpoint slice targeting a pod IP matching what's in the VM/guest 
+- Any Multus configuration can be customized, to see all the fields customizable, you can look at [NetworkSelectionElement](https://github.com/k8snetworkplumbingwg/multus-cni/blob/acfbd42719545844248ecc35cc93d0ce367f87e5/pkg/types/types.go#L128)
+
+## Non Goals
+- Altering the current behavior, the current state covers 90% of use cases. Only enhancing the Multus support by 
+providing a way to customize it, if needed, without needing KubeVirt maintainers to actively maintain it
+
+# Design
```

Comment:
Could you please add a few words about the different components introduced?
(e.g. webhook mutation admitter)

 - The webhook could be introduced from "outside" the core kubevirt, behaving as an add-on.
It will be less intrusive for most of the users who do not need it and empower the author of it to maintain it himself. If it will prove useful for more community members, we can then integrate it as a hook point into the core kubevirt.

    - Another option is to make it pluggable at build time. There may be other options to explore.

> **Note**: The option of having this as part of the kubevirt/kubevirt code-base (the webhook itself) should come with a Feature Gate.

It would also be nice to describe in more detail and formally how the "merge" algorithm works, between the annotation customized data and the multus network one.